text/0001-agile-coretime.md

```diff
@@ -0,0 +1,627 @@
+
+#### Bulk Price Progression
+
+The present proposal assumes the existence of a price-setting mechanism which takes into account several parameters:
+
+- `OLD_PRICE`: The price of the previous sale.
+- `BULK_TARGET`: the target number of cores to be purchased as Bulk Coretime Regions or renewed during the previous sale.
+- `BULK_LIMIT`: the maximum number of cores which could have been purchased/renewed during the previous sale.
+- `CORES_SOLD`: the actual number of cores purchased/renewed in the previous sale.
+- `SELLOUT_PRICE`: the price at which the most recent Bulk Coretime was purchased (*not* renewed) prior to selling more cores than `BULK_TARGET` (or immediately after, if none were purchased before). This may not have a value if no Bulk Coretime was purchased.
+
+In general we would expect the price to increase the closer `CORES_SOLD` gets to `BULK_LIMIT` and to decrease the closer it gets to zero. If it is exactly equal to `BULK_TARGET`, then we would expect the price to remain the same.
+
+In the edge case that no cores were purchased yet more cores were sold (through renewals) than the target, then we would also avoid altering the price.
+
+A simple example of this would be the formula:
+
+```
+IF SELLOUT_PRICE == NULL AND CORES_SOLD > BULK_TARGET THEN
+    RETURN OLD_PRICE
+END IF
+EFFECTIVE_PRICE := IF CORES_SOLD > BULK_TARGET THEN
+    SELLOUT_PRICE
+ELSE
+    OLD_PRICE
+END IF
+NEW_PRICE := IF CORES_SOLD < BULK_TARGET THEN
+    EFFECTIVE_PRICE * MAX(CORES_SOLD, 1) / BULK_TARGET
+ELSE
+    EFFECTIVE_PRICE + EFFECTIVE_PRICE *
+        (CORES_SOLD - BULK_TARGET) / (BULK_LIMIT - BULK_TARGET)
+END IF
+```
+
+This exists only as a trivial example to demonstrate a basic solution exists, and should not be intended as a concrete proposal.
+
+#### Intra-Leadin Price-decrease
+
+During the Leadin Period of a sale, the effective price starts higher than the Sale Price and falls to end at the Sale Price at the end of the Leadin Period. The price can thus be defined as a simple factor above one on which the Sale Price is multiplied. A function which returns this factor would accept a factor between zero and one specifying the portion of the Leadin Period which has passed.
+
+Thus we assume `SALE_PRICE`, then we can define `PRICE` as:
+
+```
+PRICE := SALE_PRICE * FACTOR((NOW - LEADIN_BEGIN) / LEADIN_PERIOD)
+```
+
+We can define a very simple progression where the price decreases monotonically from double the Sale Price at the beginning of the Leadin Period.
+
+```
+FACTOR(T) := 2 - T
+```
+
+#### Instantaneous Price Progression
+
+This proposal assumes the existence of a Relay-chain-based price-setting mechanism for the Instantaneous Coretime Market which alters from block to block, taking into account several parameters: the last price, the size of the Instantaneous Coretime Pool (in terms of cores per block) and the amount of Instantaneous Coretime waiting for processing (in terms of Core-blocks queued).
```

Comment:
```suggestion
This proposal assumes the existence of a Relay-chain-based price-setting mechanism for the Instantaneous Coretime Market which alters from block to block, taking into account several parameters: the last price, the size of the Instantaneous Coretime Pool (in terms of cores per relay-chain block) and the amount of Instantaneous Coretime waiting for processing (in terms of Core-blocks queued).
```

"core blocks queued", wouldn't "tasks queued" be better?

Reply:
One task (i.e. one parachain ID) may have more than one instance in the pool, so it's not *tasks*, but one-block cores, or a single atomic unit of Coretime. It's the resource R in the sentence "Pay the Instantaneous Coretime price for your task to be executed by R".